src/Frontend/FrontendServiceProvider.php

```diff
@@ -54,6 +54,48 @@ public function register()
                 $frontend->content($container->make(Content\CorePayload::class));
                 $frontend->content($container->make(Content\Meta::class));
 
+                $frontend->content(function (Document $document) use ($container) {
+                    $filesystem = $container->make('filesystem')->disk('flarum-assets');
+
+                    $fontawesome_preloads = [
+                        [
+                            'href' => $filesystem->url('fonts/fa-solid-900.woff2'),
+                            'as' => 'font',
+                            'type' => 'font/woff2',
+                            'crossorigin' => ''
+                        ], [
+                            'href' => $filesystem->url('fonts/fa-regular-400.woff2'),
+                            'as' => 'font',
+                            'type' => 'font/woff2',
+                            'crossorigin' => ''
+                        ]
+                    ];
```

Comment:
Maybe this should be a singleton? Preferably with a naming like "additional preloads"?

Reply:
To sate my ~~desire~~ need to learn PHP, what do we gain by doing that?

Reply:
By making this a singleton it would allow someone to say replace or otherwise remove these pre-loads if they wanted. For example if they replace the built-in FontAwesome with FontAwesome Pro.

Reply:
It's also clearer semantically, because we're not just including some random hard-coded values as part of implementation details.